offgridplanner/projects/views.py

```diff
@@ -1,9 +1,262 @@
+def db_nodes_to_js(request, proj_id=None, markers_only=False):
+    proj_id = None  # TODO remove this when project is fixed
+    if proj_id is not None:
+        project = get_object_or_404(Project, id=proj_id)
+        if project.user != request.user:
+            raise PermissionDenied
+    else:
+        project = Project.objects.first()  # TODO remove this when project is fixed
+        nodes = Nodes.objects.get(project=project)
+        df = pd.read_json(nodes.data) if nodes is not None else pd.DataFrame()
+        if not df.empty:
+            df = df[
+                [
+                    "latitude",
+                    "longitude",
+                    "how_added",
+                    "node_type",
+                    "consumer_type",
+                    "consumer_detail",
+                    "custom_specification",
+                    "is_connected",
+                    "shs_options",
+                ]
+            ]
+            power_house = df[df["node_type"] == "power-house"]
+            if markers_only is True:
+                if len(power_house) > 0 and power_house["how_added"].iat[0] == "manual":
+                    df = df[df["node_type"].isin(["power-house", "consumer"])]
+                else:
+                    df = df[df["node_type"] == "consumer"]
+            df["latitude"] = df["latitude"].astype(float)
+            df["longitude"] = df["longitude"].astype(float)
+            df["shs_options"] = df["shs_options"].fillna(0)
+            df["custom_specification"] = df["custom_specification"].fillna("")
+            df["shs_options"] = df["shs_options"].astype(int)
+            df["is_connected"] = df["is_connected"].astype(bool)
+            nodes_list = df.to_dict("records")
+            is_load_center = True
+            if (
+                len(power_house.index) > 0
+                and power_house["how_added"].iat[0] == "manual"
+            ):
+                is_load_center = False
+            return JsonResponse(
+                status=200,
+                content={"is_load_center": is_load_center, "map_elements": nodes_list},
+            )
+
+@require_http_methods(["POST"])
+# async def consumer_to_db(request, proj_id):
+def consumer_to_db(request, proj_id=None):
+    proj_id = None # TODO remove this when project is fixed
+    if proj_id is not None:
+        project = get_object_or_404(Project, id=proj_id)
+        if project.user != request.user:
+            raise PermissionDenied
+    else:
+        project = Project.objects.first() # TODO remove this when project is fixed
+        data = json.loads(request.body)
+        print(data["map_elements"])
+
+        df = pd.DataFrame.from_records(data["map_elements"])
+        if df.empty is True:
+            Nodes.objects.filter(project=project).delete()
+            return
+        df = df.drop_duplicates(subset=['latitude', 'longitude'])
+        drop_index = df[df['node_type'] == 'power-house'].index
+        if drop_index.__len__() > 1:
```

Comment:
```suggestion
        if len(drop_index) > 1:
```
I see no reason to call the magic method directly here? Seems like bad practice?